src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,214 @@
+                   "cleanup:#{cleanup} userdata:#{userdata} result:#{result}")
+          result
+        end
+
+
+        def create_pre(config_name, description, cleanup, userdata)
+          result = dbus_object.CreatePreSnapshot(config_name, escape(description), escape(cleanup),
+                                                  escape(userdata)).first
+          log.debug("create_pre_snapshot config_name:#{config_name} description:#{description} "\
+                   "cleanup:#{cleanup} userdata:#{userdata} result: #{result}")
+
+          result
+        end
+
+
+        def create_post(config_name, prenum, description, cleanup, userdata)
+          result = dbus_object.CreatePostSnapshot(config_name, prenum, escape(description),
+                                                   escape(cleanup), escape(userdata)).first
+          log.debug("create_post_snapshot config_name:#{config_name} prenum:#{prenum} "\
+                   "description:#{description} cleanup:#{cleanup} userdata:#{userdata}"\
+                   "result #{result}")
+
+          result
+        end
+
+
+        def delete(config_name, nums)
+          result = dbus_object.DeleteSnapshots(config_name, nums).first
+          log.debug("delete_snapshots config_name:#{config_name} nums:#{nums} result:#{result}")
+
+          result
+        end
+
+
+        def set_snapshot(config_name, num, description, cleanup, userdata)
+          result = dbus_object.SetSnapshot(config_name, num, escape(description), escape(cleanup),
+                                            escape(userdata)).first
+          log.debug("set_snapshot config_name:#{config_name} num:#{num} "\
+                   "description:#{description} cleanup:#{cleanup} userdata:#{userdata} "\
+                   "result #{result}")
+
+          result
+        end
+
+
+        def get_mount_point(config_name, num)
+          result = dbus_object.GetMountPoint(config_name, num).first
+          log.debug("get_mount_point config_name:#{config_name} num:#{num} result#{result}")
+
+          result
+        end
+
+
+        def list_configs
+          result = dbus_object.ListConfigs().first
+          log.debug("list_configs result:#{result}")
+
+          result.map(&:first)
+        end
+
+        def get_config(config_name)
+          result = dbus_object.GetConfig(config_name).first
+          log.debug("get_config for name '#{config_name}' result:#{result}")
+
+          result
+        end
+
+        def list_snapshots(config_name = nil)
+          result = dbus_object.ListSnapshots(config_name).first
+          log.debug("list_snapshots for name #{config_name} result:#{result}")
+          result
+        end
+
+        def dbus_object
+          @dbus_object ||= DBus::SystemBus.instance().
+            service("org.opensuse.Snapper").
+            object("/org/opensuse/Snapper")
+          @dbus_object.default_iface = "org.opensuse.Snapper"
+          @dbus_object.introspect()
+          @dbus_object
```

Comment:
this looks wrong for me as it correctly instanciate it once, then call call everytime default_iface and introspect ( and introspect is usually quite expensive ).

So I expect something like:
```
return @dbus_object if @dbus_object

@dbus_object = DBus::SystemBus.instance().
    service("org.opensuse.Snapper").
    object("/org/opensuse/Snapper")
@dbus_object.default_iface = "org.opensuse.Snapper"
@dbus_object.introspect()
@dbus_object
```